### PR TITLE
IRV classifier

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds in:
Implementation of IRV transformer(from ECFP to IRV features), IRV classifier(a learnable K-nearest neighbour method). Performances and test scripts are going to be added.

Results on tox21(10 epochs):
Train: 0.811
Valid: 0.767
about the same level of vanilla multitask network

### CODE REVIEW

**Comment by rbharath** [comment]
@miaecle Thanks for getting this in! IRV does quite well on Tox21. Let me know once this is ready to merge (it looks like you're currently running afoul of the new automatic style guide. Check in with me or @lilleswing offline if you're having trouble getting the code to style properly with it).

**Comment by miaecle** [comment]
@lilleswing Thanks for getting the style check in! I have started playing around with yapf, really a good move to get everything formatted consistently. 
Also this PR will hopefully add qm7 back into the benchmark suite. Should be done by this weekend
@rbharath Thanks for the explanation! BTW do you know what kind of tests should I add into the test script(For IRV classifier and transformer)?

File: deepchem/trans/transformers.py [hunk 1]
```diff
@@ -607,3 +607,56 @@ def transform(self, dataset):
...
+    """
+    self.X = dataset.X
+    self.n_tasks = n_tasks
+    self.K = K
+    self.y = dataset.y
+    self.w = dataset.w
+    self.transform_x = transform_x
+    self.transform_y = transform_y
+
+  def realize(self, similarity, y, w):
```
Comment (rbharath) [inline deepchem/trans/transformers.py]: Could you add doc-strings for `realize()` and `X_transform()` along with a sample usage example in the docs for `IRVTransformer`? (I believe some of the other classes in `transformers.py` have usage examples in their docs)

**Comment by rbharath** [comment]
@miaecle Thanks for cleaning up the PR and adding qm7 back in. I've added a request for more docs in the class above. Otherwise looks good (since IRV is a k-nearest neighbors variant, it's not clear an overfit test would work well. I think it's just fine to incorporate it into the benchmarks for now and worry about tests later).

**Comment by miaecle** [comment]
@rbharath Thanks for the suggestion, docs added!
I have included two tests of IRV classifier(should be easy to pass), and modified benchmark script(adding  qm7b, rf_regression, IRV and -t arg). qm7 and qm7b now directly use the user-defined features(from .mat file), performance difference should come from different hyperparameters(no need to further optimize since it is the default settings of benchmark).
Let me know if anything looks wrong or needs modification!

**Comment by rbharath** [comment]
@miaecle Looks great! Thanks for adding in the RF and the QM7B Benchmarks. Would it make sense to add a MAE score for QM7B to match the QM7 MAE numbers?

**Comment by miaecle** [comment]
@rbharath qm7b has 14 tasks, each in different range(on different order of magnitude). Hence I think average of mae is not that meaningful for the dataset, mae for each individual task may worth comparing though.

**Comment by rbharath** [comment]
@miaecle Ah, good point. Let's discuss offline this week with @joegomes to figure out a strategy that makes sense for qm7b. This PR looks good! 

LGTM.

**Pull request merged**